profiling/test_import_and_query.py

```diff
@@ -0,0 +1,190 @@
+# Download the datasets at
+#  - https://storage.googleapis.com/ann-datasets/ann-benchmarks/sift-128-euclidean.hdf5
+#  - https://storage.googleapis.com/ann-datasets/ann-benchmarks/dbpedia-openai-1000k-angular.hdf5
+#  and place it in the same folder as this file.
+#
+# In addition to weavaite, you will need to install the "h5py" package.
+#
+# To run all tests use `pytest -s test_sift.py` in the profiling directory. If you want to run only one of the tests
+# you can use the `-k` flag, for example `pytest -s test_sift.py -k test_sift_v3`
+
+EF_VALUES = [16, 32, 64, 128, 256, 512]
+LIMIT = 10
+
+
+def create_schema(
+    client: weaviate.WeaviateClient, name: str, efC: int, m: int, shards: int, distance: str
+) -> Collection:
+    client.collections.delete(name)
+    vi = wvc.config.Configure.VectorIndex.hnsw(
+        ef_construction=efC,
+        max_connections=m,
+        ef=-1,
+        distance_metric=wvc.config.VectorDistances(distance),
+    )
+
+    return client.collections.create(
+        name=name,
+        vectorizer_config=wvc.config.Configure.Vectorizer.none(),
+        vector_index_config=vi,
+        properties=[
+            wvc.config.Property(
+                name="i",
+                data_type=wvc.config.DataType.INT,
+            )
+        ],
+        inverted_index_config=wvc.config.Configure.inverted_index(index_timestamps=False),
+        sharding_config=wvc.config.Configure.sharding(desired_count=shards),
+    )
+
+
+def load_records_v4(collection: Collection, vectors: List[List[float]]) -> None:
+    start = time.time()
+    with collection.batch.fixed_size(1000) as batch:
+        for i, vector in enumerate(vectors):
+            data_object = {"i": i}
+
+            batch.add_object(
+                properties=data_object,
+                vector=vector,
+                uuid=uuid.UUID(int=i),
+            )
+
+    print(f"V4: Finished writing {len(vectors)} records in {time.time()-start}s")
+
+
+def load_records_v3(client: weaviate.Client, vectors: List[List[float]], name: str) -> None:
+    start = time.time()
+
+    client.batch.configure(batch_size=1000, num_workers=2)
+
+    with client.batch as batch:
+        for i, vector in enumerate(vectors):
+            data_object = {"i": i}
+
+            batch.add_data_object(
+                data_object=data_object,
+                vector=vector,
+                class_name=name,
+                uuid=uuid.UUID(int=i),
+            )
+
+    print(f"V3: Finished writing {len(vectors)} records in {time.time()-start}s")
+
+
+def query_v4(
+    collection: Collection, vectors: List[List[float]], neighbours: List[List[int]], ef: int
+) -> None:
+    collection.config.update(vector_index_config=wvc.config.Reconfigure.VectorIndex.hnsw(ef=ef))
+    start = time.time()
+    recall = 0.0
+
+    for i, vec in enumerate(vectors):
+        objs = collection.query.near_vector(
+            near_vector=list(vec), limit=LIMIT, return_properties=[]
```

Comment:
Is this `list()` required here? Does it impact performance at all? I only notice as it's not present in the v3 query

Reply:
yes, near vector does not have the get_vector wrapper - however I am not sure if we should add it as that would affect the runtime speed for all queries